src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -106,6 +118,15 @@ public override void VisitNamespace(INamespaceSymbol symbol)
 				}
 			}
 
+			private static void ReportDiagnostic(GeneratorExecutionContext context, Diagnostic diagnostic)
+			{
+#if NETFRAMEWORK
+				throw new InvalidOperationException(diagnostic.GetMessage());
+#else
+				context.ReportDiagnostic(diagnostic);
+#endif
+			}
```

Comment:
It would make sense, though the semantic is not the same, where `ReportDiagnostic` does not stop the processing on call sites. It's not a problem but it needs to be handled explicitly at the call site.

Reply:
> though the semantic is not the same

Good point.

The alternative is introducing an extension method in Uno, for example, `ReportDiagnosticOrThrow` which makes the intent clear, and it will be a single location where we have the `#if`

Let me know which approach you like more.

Reply:
I'll probably discard https://github.com/unoplatform/Uno.SourceGeneration/pull/155 and keep the approach in this PR.

Later if I worked on this for other generators, I'll introduce a common `ReportDiagnosticOrThrow` extension.

Reply:
Let's do this, thanks.